libraries/plugins/custom_operations/custom_evaluators.cpp

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (c) 2019 oxarbitrage and contributors.
+ *
+ * The MIT License
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+#include <graphene/chain/database.hpp>
+
+#include <graphene/custom_operations/custom_operations_plugin.hpp>
+#include <graphene/custom_operations/custom_objects.hpp>
+#include <graphene/custom_operations/custom_evaluators.hpp>
+
+namespace graphene { namespace custom_operations {
+
+custom_generic_evaluator::custom_generic_evaluator(database& db, const account_id_type account)
+{
+   _db = &db;
+   _account = account;
+}
+
+vector<object_id_type> custom_generic_evaluator::do_apply(const account_storage_map& op)
+{
+   auto &index = _db->get_index_type<account_storage_index>().indices().get<by_account_catalog_key>();
+   vector<object_id_type> results;
+
+   if (op.extensions.value.remove.valid() && *op.extensions.value.remove)
+   {
+      for(auto const& row: *op.extensions.value.key_values) {
+         auto itr = index.find(make_tuple(_account, *op.extensions.value.catalog, row.first));
+         if(itr != index.end()) {
+            results.push_back(itr->id);
+            _db->remove(*itr);
+         }
+      }
+   }
+   else {
+      for(auto const& row: *op.extensions.value.key_values) {
+         if(row.first.length() > CUSTOM_OPERATIONS_MAX_KEY_SIZE)
+         {
+            wlog("Key can't be bigger than ${max} characters", ("max", CUSTOM_OPERATIONS_MAX_KEY_SIZE));
```

Comment:
Please make this a `dlog`. Node operators probably won't care about it, and they can't do anything about it either. Clients might be interested but will not see the message.

Reply:
changed at https://github.com/bitshares/bitshares-core/pull/1926/commits/0f8560be0186f0323bc456683fdc3ec3d06e86b3